### PR TITLE
[CINN]Support full broadcast

### PR DESCRIPTION
**Pull request by phlrain**
### PR types
 Function optimization
### PR changes

Others
### Description

Pcard-67164 

支持 shape = [1]的算子 broadcast 成 [S0, 32, S2]的格式

### CODE REVIEW

**Review by BiynXu — APPROVED** [review]

File: paddle/cinn/hlir/dialect/operator/transforms/cinn_group_cluster_pass.cc [hunk 1]
```diff
@@ -527,17 +531,44 @@ void GetClusterNodeBasicInfo(::pir::Operation* op,
                            .type()
                            .dyn_cast<paddle::dialect::DenseTensorType>()
                            .dims());
+
+    pir::ShapeConstraintIRAnalysis& shape_analysis =
+        pir::ShapeAnalysisManager::Instance().Get(op->GetParentProgram());
+    if (shape_analysis.HasShapeOrDataForValue(op->operand_source(0))) {
+      auto sym_shape =
```
Comment (tc20042008) [inline paddle/cinn/hlir/dialect/operator/transforms/cinn_group_cluster_pass.cc:538]: const auto&

File: paddle/cinn/hlir/dialect/operator/transforms/cinn_group_cluster_pass.cc [hunk 2]
```diff
@@ -527,17 +531,44 @@ void GetClusterNodeBasicInfo(::pir::Operation* op,
...
   } else if (cluster_node->group_kind == cinn::hlir::framework::kElementWise) {
     cluster_node->loop_ranges =
         phi::vectorize(op->result(0)
                            .type()
                            .dyn_cast<paddle::dialect::DenseTensorType>()
                            .dims());
+    pir::ShapeConstraintIRAnalysis& shape_analysis =
+        pir::ShapeAnalysisManager::Instance().Get(op->GetParentProgram());
+    if (shape_analysis.HasShapeOrDataForValue(op->result(0))) {
+      auto sym_shape =
```
Comment (tc20042008) [inline paddle/cinn/hlir/dialect/operator/transforms/cinn_group_cluster_pass.cc:563]: const auto&

File: paddle/cinn/hlir/dialect/operator/transforms/cinn_group_cluster_pass.cc [hunk 3]
```diff
@@ -579,6 +610,22 @@ void GetClusterNodeBasicInfo(::pir::Operation* op,
       return broadcast_axes;
     }();
     sch_node->factor_info = output_shape;
+
+    pir::ShapeConstraintIRAnalysis& shape_analysis =
+        pir::ShapeAnalysisManager::Instance().Get(op->GetParentProgram());
+    if (shape_analysis.HasShapeOrDataForValue(op->result(0))) {
+      auto sym_shape =
```
Comment (tc20042008) [inline paddle/cinn/hlir/dialect/operator/transforms/cinn_group_cluster_pass.cc:617]: const auto&

File: paddle/cinn/hlir/dialect/operator/transforms/replace_dynamic_expand_pass.cc [hunk 4]
```diff
@@ -80,6 +80,20 @@ class DynamicExpandOpPattern
         broadcast->result(0),
         shape_analysis.GetShapeOrDataForValue(op.result(0)));
 
+    if (auto pre_full = broadcast->operand_source(0)
+                            .defining_op()
+                            ->dyn_cast<paddle::dialect::FullOp>()) {
+      auto input_dim = pre_full.result(0)
```
Comment (tc20042008) [inline paddle/cinn/hlir/dialect/operator/transforms/replace_dynamic_expand_pass.cc:86]: const auto&

**Pull request merged**